bagit.py

```diff
@@ -579,9 +585,50 @@ def files_to_be_fetched(self):
         local filename
         """
 
-        for url, file_size, filename in self.fetch_entries():
+        for _, _, filename in self.fetch_entries():
             yield filename
 
+    def fetch(self, force=False):
+        """
+        Fetches files from the fetch.txt
+
+        Arguments:
+            force (boolean): Fetch files even when they are present in the data directory
+        """
+        proxy_handler = ProxyHandler() # will default to adhere to *_proxy env vars
+        opener = build_opener(proxy_handler)
+        user_agent = "bagit.py/%s (Python/%s)" % (VERSION, sys.version_info)
+        for url, expected_size, filename in self.fetch_entries():
+            if not fnmatch_any(url, self.fetch_url_whitelist):
+                raise BagError(_("Malformed URL in fetch.txt: %s, matches none of the whitelisted URL patterns %s") % (url, self.fetch_url_whitelist))
+            expected_size = -1 if expected_size == '-' else int(expected_size)
+            if filename in self.payload_files() and not force:
+                LOGGER.info(_("File already fetched: %s"), filename)
+                continue
+            req = Request(url)
+            req.add_header('User-Agent', user_agent)
+            resp = opener.open(req)
+            headers = resp.info()
+            if expected_size >= 0:
+                if "content-length" not in headers:
```

Comment:
This would only apply to HTTP requests but the intention of this code is to support other protocols, which suggests that this would need to be a conditional warning since I don't believe the stdlib sets a header like that for other protocols.